stdlib/public/core/UnsafeRawPointer.swift.gyb

```diff
@@ -0,0 +1,557 @@
+//===--- UnsafeRawPointer.swift.gyb --------------------------*- swift -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+%import gyb
+
+% for mutable in (True, False):
+%  Self = 'UnsafeMutableRawPointer' if mutable else 'UnsafeRawPointer'
+%  a_Self = 'an `UnsafeMutableRawPointer`' if mutable else 'an `UnsafeRawPointer`'
+%  Mutable = 'Mutable' if mutable else ''
+
+/// A raw pointer for accessing untyped data. This provides no
+/// automatic memory management, no type safety, and no alignment
+/// guarantees. This implements Strideable to provide a view
+/// of byte-addressable memory.
+@_fixed_layout
+public struct Unsafe${Mutable}RawPointer : Strideable, Hashable, _Pointer {
+  /// The underlying raw pointer.
+  /// Implements conformance to the public protocol `_Pointer`.
+  public let _rawValue: Builtin.RawPointer
+
+  /// Convert a builtin raw pointer to a Unsafe${Mutable}RawPointer.
+  @_transparent
+  public init(_ _rawValue: Builtin.RawPointer) {
+    self._rawValue = _rawValue
+  }
+
+  /// Convert an opaque pointer to ${a_Self}.
+  @_transparent
+  public init(_ other: OpaquePointer) {
+    _rawValue = other._rawValue
+  }
+
+  /// Convert an opaque pointer to ${a_Self}.
+  ///
+  /// Returns nil if `from` is nil.
+  @_transparent
+  public init?(_ other: OpaquePointer?) {
+    guard let unwrapped = other else { return nil }
+    _rawValue = unwrapped._rawValue
+  }
+
+  /// Construct ${a_Self} with a given pattern of bits.
+  @_transparent
+  public init?(bitPattern: Int) {
+    if bitPattern == 0 { return nil }
+    _rawValue = Builtin.inttoptr_Word(bitPattern._builtinWordValue)
+  }
+
+  /// Construct ${a_Self} with a given pattern of bits.
+  @_transparent
+  public init?(bitPattern: UInt) {
+    if bitPattern == 0 { return nil }
+    _rawValue = Builtin.inttoptr_Word(bitPattern._builtinWordValue)
+  }
+
+  /// Convert an Unsafe${Mutable}Pointer ${a_Self}.
```

Comment:
I'll rewrite all the initializer comments to read "Convert <something> to ${a_Self}."